src/graphnet/components/loss_functions.py

```diff
@@ -61,6 +61,19 @@ def _forward(self, prediction: Tensor, target: Tensor) -> Tensor:
         """Syntax similar to `.forward` for implentation in inheriting classes."""
 
 
+class GaussLoss(LossFunction):
+    """Gaussian negative log likelihood loss."""
+
+    def _forward(self, prediction: Tensor, target: Tensor) -> Tensor:
+        """Implementation of loss calculation."""
+
+        variance = 1 / prediction[:, 2]
+
+        loss = torch.nn.GaussianNLLLoss()
+
+        return loss(prediction, target, variance, eps=1e-06, reduction="mean")
```

Comment:
Making sure that we are returning the loss "element"/term for each example in the batch rather than averaging over them. (This is necessary for applying per-event weights.)
```suggestion
        loss = torch.nn.GaussianNLLLoss()

        variance = 1.0 / prediction[:, 2]
        elements = loss(prediction, target, variance, eps=1e-06, reduction="none")
        elements = torch.sum(elements, dim=-1)  # Sum along coordinate dimension
        return elements
```